### DISCUSSION TITLE
GetObject result.ContentLength different from Response Content

### DISCUSSION THREAD

**Discussion by gautampunhani** [aws/aws-sdk-go-v2#2525] [Q&A] [answered]
### Describe the bug

s3 Content Length and go-sdk's ContentLength is different. This causes our endpoints which were streaming data to fail. This has started happening after we did upgrades mentioned here #2370 `go get -u github.com/aws/aws-sdk-go-v2/...` 
<img width="887" alt="Screenshot 2024-03-01 at 6 30 42 PM" src="https://github.com/aws/aws-sdk-go-v2/assets/2483883/8b88f441-c380-4fc2-a604-2cdcd88ba95d">

### Expected Behavior

Content length should be same as received by the Http call in Debug logs

### Current Behavior

Actual ContentLength : 25676
GetObject's `result.ContentLength` : 824635599248

### Reproduction Steps

Please refer https://github.com/gautampunhani/aws-sdk-content-length-issue/tree/main , Which tries to access an AWS Open Data S3 file.

### AWS Go SDK V2 Module Versions Used

Please refer here for versions used : https://github.com/gautampunhani/aws-sdk-content-length-issue/blob/main/go.mod

### Compiler and Version used

go version go1.21.3 

### Operating System and version

darwin/amd64

**Comment by RanVaknin** [comment, marked as answer]
Hi @gautampunhani,

Thanks for reaching out. 
I don't think this has anything to do with #2370. 

`response.ContentLength` is a *int64. 
In[ your code](https://github.com/gautampunhani/aws-sdk-content-length-issue/blob/8c5f0c948741aa3c3ce67604dc6e3c6e496bb09d/main.go#L37) you are printing the pointer of content length:

```go
fmt.Printf("\n\nContent Length %d", out.ContentLength)
```

Instead, you need to dereference it:
```go
fmt.Printf("\n\nContent Length %d", *out.ContentLength)
```

Thanks,
Ran~

**Reply by gautampunhani** [reply]
Thanks Ran, We can close this discussion. Is it something mentioned already in some docs when we are bumping version?
